sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/SchemaPruningSuite.scala

```diff
@@ -460,6 +460,60 @@ abstract class SchemaPruningSuite
     checkAnswer(query4, Row(2, null) :: Row(2, 4) :: Nil)
   }
 
+  testSchemaPruning("select nested field in window function") {
+    val windowSql =
+      """
+        |with contact_rank as (
+        |  select row_number() over (partition by address order by id desc) as rank,
+        |  contacts.*
+        |  from contacts
+        |)
+        |select name.first, rank from contact_rank
+        |where name.first = 'Jane' AND rank = 1
+        |""".stripMargin
+    val query1 = sql(windowSql)
+    checkScan(query1, "struct<id:int,name:struct<first:string>,address:string>")
+    checkAnswer(query1, Row("Jane", 1) :: Nil)
```

Comment:
query1 -> query

Reply:
Good catch. `sort by` is a local sort, so I also updated this test to do repartition first to make results more predictable.

Reply:
Changed